File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int ans[]=new int[nums.length];
        int n=ans.length;
        int i=0;
        int j=n-1;
        int k=n-1;
        while(i<=j && k>=0)
        {
            if(Math.abs(nums[i]) < Math.abs(nums[j]))
            {
                int ele=Math.abs(nums[j]);
                ans[k--]=(ele*ele);
                j--;
            }else
            {
                 int ele=Math.abs(nums[i]);
                ans[k--]=(ele*ele);
                i++;
            }

        }
        return ans;
    }
}

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int>ans;
        int n=nums.size();
        cout<<n;
        int i=0;
        int j=n-1;
        int k=n-1;
        while(i<=j && k>=0)
        {
            if(abs(nums[i]) < abs(nums[j]))
            {
                int p=abs(nums[j]);
                ans.insert(ans.begin(),p*p);
                k--;
                j--;
            }else
            {
                int p=abs(nums[i]);
                ans.insert(ans.begin(),p*p);
                k--;
                i++;
            }
        }
        return ans;
    }
};